adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -24,56 +24,89 @@ import scala.concurrent.{ ExecutionContext, Future }
 class AkkaHttpAdapter private (private val options: AkkaHttpServerOptions)(implicit ec: ExecutionContext) {
   private val akkaInterpreter = AkkaHttpServerInterpreter(options)(ec)
 
-  def makeHttpService[R, E](
+  def makeHttpService[R, E](adapter: HttpAdapter[R, E])(implicit runtime: Runtime[R]): Route =
+    akkaInterpreter.toRoute(adapter.serverEndpoints[R].map(TapirAdapter.convertHttpEndpointToFuture(_)))
+
+  def makeHttpUploadService[R, E](adapter: HttpUploadAdapter[R, E])(implicit
+    runtime: Runtime[R],
+    requestCodec: JsonCodec[GraphQLRequest],
+    mapCodec: JsonCodec[Map[String, Seq[String]]]
+  ): Route =
+    akkaInterpreter.toRoute(TapirAdapter.convertHttpEndpointToFuture(adapter.serverEndpoint[R]))
+
+  def makeWebSocketService[R, E](
+    adapter: WebSocketAdapter[R, E]
+  )(implicit ec: ExecutionContext, runtime: Runtime[R], materializer: Materializer): Route =
+    akkaInterpreter.toRoute(
+      convertWebSocketEndpoint(
+        adapter
+          .serverEndpoint[R]
+          .asInstanceOf[
+            ServerEndpoint.Full[
+              Unit,
+              Unit,
+              (ServerRequest, String),
+              StatusCode,
+              (String, CalibanPipe),
+              ZioWebSockets,
+              RIO[R, *]
+            ]
+          ]
+      )
+    )
+
+  @deprecated
```

Comment:
I kept the old function but deprecated and without the `requestInterceptor`. Is it worth it?

Reply:
I got rid of it